### PR TITLE
Allow lists and dicts as msg/var parameters to debug module

### PR DESCRIPTION
**Pull request by gnosek**
##### SUMMARY
This pull request allows lists and dicts to be passed as values of the `debug` module. Strictly speaking, a similar effect may be achieved using `with_items`, but the output is clearer when implemented natively. Please see additional information for before/after comparison:

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
`plugins/action/debug.py`

##### ANSIBLE VERSION
```
ansible 2.5.0 (debug-list 4b50d22be3) last updated 2017/10/14 12:09:47 (GMT +200)
  config file = None
  configured module search path = [u'/Users/blackfire/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/blackfire/src/ansible/lib/ansible
  executable location = /Users/blackfire/src/ansible/bin/ansible
  python version = 2.7.10 (default, Jul 14 2015, 19:46:27) [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)]
```

##### ADDITIONAL INFORMATION

Before:
```
  - debug:
      var: "{{ item }}"
    with_items:
    - str1_var
    - str2_var
```
```
ok: [localhost] => (item=str1_var) => {
    "item": "str1_var",
    "str1_var": "string variable 1"
}
ok: [localhost] => (item=str2_var) => {
    "item": "str2_var",
    "str2_var": "string variable 2"
}
```

After:
```
  - debug:
      var:
      - str1_var
      - str2_var
```
```
ok: [localhost] => {
    "var": {
        "str1_var": "string variable 1",
        "str2_var": "string variable 2"
    }
}
```

### CODE REVIEW

**Review by dagwieers — APPROVED** [review]
I like this functionality, but leave it up to core developers to merge this.

**Comment by gnosek** [comment]
Hmm,

The test failures at https://app.shippable.com/github/ansible/ansible/runs/40890/33/tests show usage like `debug var={{ external_ip }}` -- this doesn't seem legal (it would evaluate to e.g. `debug var=10.0.0.1`) but if it worked before, it's probably widespread enough to cause compatibility problems. I'll see what I can do about it

**Comment by bcoca** [comment]
@gnosek that test was incorrect, but the expression is legal, you should not fail if var points to a template.

**Comment by gnosek** [comment]
@bcoca, the tests didn't fail. I had found the error you mention earlier and fixed it by reverting one commit. AFAIK the failures come from an OpenSUSE repo having problems. I fixed the tests as a cleanup but if you prefer to keep them as is, I can drop that commit.

**Comment by mkrizek** [comment]
@gnosek are you still interested in this PR? If so, would you mind rebasing and resolving conflicts? Thanks!

**Comment by Akasurde** [comment]
@gnosek Are you still working on this? Thanks.

needs_info

**Pull request closed without merging**
